server/service/src/pack_variant/update.rs

```diff
@@ -0,0 +1,77 @@
+use repository::{
+    PackVariant, PackVariantRow, PackVariantRowRepository, RepositoryError, StorageConnection,
+};
+
+use crate::{invoice_line::validate::check_item_exists, service_provider::ServiceContext};
+
+use super::validate::check_pack_variant_exists;
+
+#[derive(PartialEq, Debug)]
+pub enum UpdatePackVariantError {
+    ItemDoesNotExist,
+    PackVariantDoesNotExist,
+    DatabaseError(RepositoryError),
+}
+
+pub struct UpdatePackVariant {
+    pub id: String,
+    pub item_id: String,
+    pub short_name: String,
+    pub long_name: String,
+}
+
+pub fn update_pack_variant(
+    ctx: &ServiceContext,
+    input: UpdatePackVariant,
+) -> Result<PackVariant, UpdatePackVariantError> {
+    let pack_variant = ctx
+        .connection
+        .transaction_sync(|connection| {
+            let pack_variant_row = validate(connection, &input)?;
+            let updated_pack_variant = generate(input, pack_variant_row);
+            let repo = PackVariantRowRepository::new(&connection);
+            repo.upsert_one(&updated_pack_variant)?;
+
+            repo.find_one_by_id(&updated_pack_variant.id)?
+                .ok_or(UpdatePackVariantError::PackVariantDoesNotExist)
+        })
+        .map_err(|error| error.to_inner_error())?;
+    Ok(pack_variant)
+}
+
+impl From<RepositoryError> for UpdatePackVariantError {
+    fn from(error: RepositoryError) -> Self {
+        UpdatePackVariantError::DatabaseError(error)
+    }
+}
+
+fn validate(
+    connection: &StorageConnection,
+    input: &UpdatePackVariant,
+) -> Result<PackVariantRow, UpdatePackVariantError> {
+    let pack_variant_row = match check_pack_variant_exists(connection, &input.id)? {
```

Comment:
It's only checked in insert since I disabled pack size in update and not passing it through the api as an insert, so there won't be a variant id to check against... 

Don't think we should allow them to change the pack size in an already existing variant...